src/seedu/addressbook/data/person/Address.java

```diff
@@ -23,19 +40,56 @@
     public Address(String address, boolean isPrivate) throws IllegalValueException {
         String trimmedAddress = address.trim();
         this.isPrivate = isPrivate;
-        if (!isValidAddress(trimmedAddress)) {
-            throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
+        String[] decodedAddress = decodeAddress(trimmedAddress);
+        if (decodedAddress.length != 4) {
+        	throw new IllegalValueException(MESSAGE_FOR_INVALID_ADDRESS);
         }
-        this.value = trimmedAddress;
+        this.block = new Block(decodedAddress[BLOCK_INDEX]);
+        this.street = new Street(decodedAddress[STREET_INDEX]);
+        this.unit = new Unit(decodedAddress[UNIT_INDEX]);
+        this.postalCode = new PostalCode(decodedAddress[POSTAL_CODE_INDEX]);
+        value = trimmedAddress;
+    }
+    
+    /**
+     * split input address string into block, street, unit and postal code
+     * 
+     * @param string represents full address
+     * @return string array of block, street, unit and postal code
+     */
+    private String[] decodeAddress(String encodedAddress) {
+    	return encodedAddress.split(", ");
     }
 
     /**
-     * Returns true if a given string is a valid person email.
+     * Returns true if a given string is a valid address.
      */
-    public static boolean isValidAddress(String test) {
-        return test.matches(ADDRESS_VALIDATION_REGEX);
+    public static boolean isValidAddress(String[] decodedAddress) {
+        if (decodedAddress.length != NUMBER_OF_PROPERTIES_IN_ADDRESS) {
+        	return false;
+        }
+        return Block.isValidBlock(decodedAddress[BLOCK_INDEX])
+        		&& Street.isValidStreet(decodedAddress[STREET_INDEX])
+        		&& Unit.isValidUnit(decodedAddress[UNIT_INDEX])
+        		&& PostalCode.isValidPostalCode(decodedAddress[POSTAL_CODE_INDEX]);
     }
 
+    public Block getBlock() {
```

Comment:
Consider adding all of these to the toString() method and allowing the user to address as a whole rather than small components.